libraries/AC_AttitudeControl/AC_WeatherVane.cpp

```diff
@@ -159,13 +166,17 @@ bool AC_WeatherVane::get_yaw_out(float &yaw_output, const int16_t pilot_yaw, con
     const float deadzone_cdeg = _min_dz_ang_deg*100.0;
     float output = 0.0;
     const char* dir_string = "";
+
+    // should we enable pitch input for nose-in and tail-in?
+    const bool pitch_enable = (uint8_t(_options.get()) & uint8_t(Options::PITCH_ENABLE)) != 0;
+
     switch (dir) {
         case Direction::OFF:
             reset();
             return false;
 
         case Direction::NOSE_IN:
-            if (is_positive(pitch_cdeg)) {
+            if (pitch_enable && is_positive(pitch_cdeg - deadzone_cdeg)) {
                 output = fabsf(roll_cdeg) + pitch_cdeg;
```

Comment:
```suggestion
                output = fabsf(roll_cdeg) + pitch_cdeg - deadzone_cdeg;
```
Needs to match the `is_positive` amount so there is no step change when you move from using pitch to not.

Reply:
thanks, fixed (in 2 places)